### ISSUE TITLE
Docker service not coming up on centos 7.5 with kops 1.11

### ISSUE THREAD

**Issue by apurvara** [kubernetes/kops#6348]
**1. What `kops` version are you running? The command `kops version`, will display
 this information.**
Kops 1.11

**2. What Kubernetes version are you running? `kubectl version` will print the
 version if a cluster is running or provide the Kubernetes version specified as
 a `kops` flag.**
Kubectl version is 1.11.6 

**3. What cloud provider are you using?**
AWS

**4. What commands did you run?  What is the simplest way to reproduce this issue?**
kops create cluster with all the required flag (same flags which works with kops 1.10.1 on same configuation)
Failed Cluster install was with Kops 1.11 on CentOS 7.5 AMI (with SELinux disabled)

**5. What happened after the commands executed?**
Kubernetes API service did not came up.. master instance on the ELB was out of service  

**6. What did you expect to happen?**
Cluster to come up

**9. Anything else do we need to know?**
On debugging of master and nodes found that docker service was not coming up. Docker 17.03.2  version was installed on the nodes and master. and on similar configuration with Kops 1.10.1 everything is working fine.

**Comment by allcloud-jonathan** [comment]
Can you check `journal -xe` for output indicating `nodeup` failures?

**Comment by zetaab** [comment]
the problem with centos7 is that it does not support overlay2 volume driver.

**Comment by apurvara** [comment]
> the problem with centos7 is that it does not support overlay2 volume driver.

How to solve ... as a policy i can use only redhat7 or centos7 in my org

**Comment by dignajar** [comment]
I have the same problem.

After install the cluster via Kops on Centos 7.5.

```
rpm -qa | grep docker
docker-ce-17.03.2.ce-1.el7.centos.x86_64
docker-ce-selinux-17.03.2.ce-1.el7.centos.noarch
```

```
/usr/bin/dockerd --ip-masq=false --iptables=false --log-driver=json-file --log-level=warn --log-opt=max-file=5 --log-opt=max-size=10m --storage-driver=overlay2

WARN[0000] containerd: low RLIMIT_NOFILE changing to max  current=1024 max=4096
Error starting daemon: error initializing graphdriver: driver not supported
```

**Comment by dignajar** [comment]
I solved the problem to deploy with Kops 1.11.0 on Centos 7.5.

```
$ uname -ar

Linux ip-172-16-1-68.node.pmiint.ocean 3.10.0-862.3.2.el7.x86_64 #1 SMP Mon May 21 23:36:36 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
```

```
$ docker version

Client:
 Version:      17.09.0-ce
 API version:  1.32
 Go version:   go1.8.3
 Git commit:   afdb6d4
 Built:        Tue Sep 26 22:41:23 2017
 OS/Arch:      linux/amd64

Server:
 Version:      17.09.0-ce
 API version:  1.32 (minimum version 1.12)
 Go version:   go1.8.3
 Git commit:   afdb6d4
 Built:        Tue Sep 26 22:42:49 2017
 OS/Arch:      linux/amd64
 Experimental: false
```

```
$ kops validate cluster
Using cluster from kubectl context: <CLUSTER NAME>

Validating cluster <CLUSTER NAME>

INSTANCE GROUPS
NAME			ROLE	MACHINETYPE	MIN	MAX	SUBNETS
master-eu-central-1a	Master	m5.large	1	1	eu-central-1a
nodes			Node	m5.large	1	1	eu-central-1a

NODE STATUS
NAME						ROLE	READY
ip-172-----.eu-central-1.compute.internal	master	True
ip-172-----.eu-central-1.compute.internal	node	True
```

In the configuration yaml of Kops you need to add the docker version and storage parameter:
```
  docker:
    version: 17.09.0
    storageOpts:
      - "overlay2.override_kernel_check=1"
```

One problem I had been the dependencies with `container-selinux`, I installed by hand in the instances.

In the source code is listed but not sure why is not installed.
https://github.com/kubernetes/kops/blob/806a4f0c1a9991d08fec2a5c3ecfba06683f7cf3/nodeup/pkg/model/docker.go#L504-L524

Regards

**Comment by zetaab** [comment]
solved in #6491
